control-plane/pkg/apis/internals/kafka/eventing/v1alpha1/consumer_group_types.go

```diff
@@ -156,12 +156,13 @@ func (c *ConsumerGroup) GetStatus() *duckv1.Status {
 func (cg *ConsumerGroup) ConsumerFromTemplate(options ...ConsumerOption) *Consumer {
 	// TODO figure out naming strategy, is generateName enough?
 	c := &Consumer{
-		ObjectMeta: cg.Spec.Template.ObjectMeta,
-		Spec:       cg.Spec.Template.Spec,
+		ObjectMeta: *cg.Spec.Template.ObjectMeta.DeepCopy(),
+		Spec:       *cg.Spec.Template.Spec.DeepCopy(),
```

Comment:
Why are we switching to using deep copies here?

Reply:
because we're changing the fields down below and if objects come from informers they must not be modified, so "it's better to be safe than sorry"